actors/builtin/miner/miner_actor.go

```diff
@@ -544,21 +546,18 @@ func (a Actor) ProveCommitSector(rt Runtime, params *ProveCommitSectorParams) *a
 
 	store := adt.AsStore(rt)
 	var st State
-	rt.State().Readonly(&st)
-
-	// Verify locked funds are are at least the sum of sector initial pledges.
-	// Note that this call does not actually compute recent vesting, so the reported locked funds may be
-	// slightly higher than the true amount (i.e. slightly in the miner's favour).
-	// Computing vesting here would be almost always redundant since vesting is quantized to ~daily units.
-	// Vesting will be at most one proving period old if computed in the cron callback.
-	verifyPledgeMeetsInitialRequirements(rt, &st)
-
+	var precommit *SectorPreCommitOnChainInfo
 	sectorNo := params.SectorNumber
-	precommit, found, err := st.GetPrecommittedSector(store, sectorNo)
-	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load pre-committed sector %v", sectorNo)
-	if !found {
-		rt.Abortf(exitcode.ErrNotFound, "no pre-committed sector %v", sectorNo)
-	}
+	rt.State().Transaction(&st, func() {
+		verifyPledgeMeetsInitialRequirements(rt, &st)
```

Comment:
This is dropping the comment about verify funds. I believe that information is still correct and relevant. The comment should move to the `verifyPledgeMeetsIntialRequirements` declaration.

Reply:
We no longer vest IPs so I believe the whole comment is now incorrect and misleading.

Reply:
I think there's still something to call out here.
- It's still true that this doesn't compute vesting (keep it that way)
- A miner could be in IP debt but have locked funds(*), and this won't unlock them
- So it's not in the miner's favour any more

(*) Maybe this is in fact impossible in the current code, but if so, it's a brittle condition depending on the implementation of AwardBlockReward, and could easily change.